openwisp_utils/tests/test_ci_failure_bot.py

```
@@ -0,0 +1,366 @@
+        bot = CIFailureBot()
+        mock_workflow_run = Mock()
+        mock_job = Mock()
+        mock_job.conclusion = "success"
+        mock_workflow_run.jobs.return_value = [mock_job]
+        self.mock_repo.get_workflow_run.return_value = mock_workflow_run
+        logs = bot.get_build_logs()
+        self.assertEqual(logs, [])
+
+    @patch("openwisp_utils.ci_failure_bot.subprocess.run")
+    @patch("openwisp_utils.ci_failure_bot.requests.get")
+    def test_get_pr_diff_success(self, mock_requests, mock_subprocess):
```

Ruff (0.14.14)

[warning] Unused method argument: `mock_requests`

(ARG002)